lazy_dataset/core.py

```diff
@@ -1383,7 +1383,7 @@ def apply(self, apply_fn: callable) -> 'Dataset':
             apply_fn: For now, it is a single function, e.g.,
                 `lambda ds: ds.shard(num_shards, shard_index)`   but can
                 potentially be a list in future implementations.
-
+            lazy:
```

Comment:
Can you add a comment here what `lazy=True` does? At least for me, it is not immediately clear that this applies the function multiple times, once for every iteration. Or at least a pointer to `ApplyDataset`